### PR TITLE
Downgrade klauspost/compress to fix data corruption at max. compression

### PR DESCRIPTION
**Pull request by MichaelEischer**
What does this PR change? What problem does it solve?
-----------------------------------------------------
Switch back to the version of the `compress` library used by restic 0.16.2.

Was the change previously discussed in an issue or on the forum?
----------------------------------------------------------------
Temporary workaround for https://github.com/restic/restic/issues/4677 .

Checklist
---------

- [x] I have read the [contribution guidelines](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#providing-patches).
- [x] I have [enabled maintainer edits](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added tests for all code changes.
- [ ] I have added documentation for relevant changes (in the manual).
- [ ] There's a new file in `changelog/unreleased/` that describes the changes for our users (see [template](https://github.com/restic/restic/blob/master/changelog/TEMPLATE)).
- [x] I have run `gofmt` on the code in all commits.
- [x] All commit messages are formatted in the same style as [the other commits in the repo](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#git-commits).
- [x] I'm done! This pull request is ready for review.

### CODE REVIEW

File: changelog/unreleased/issue-4677 [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+level (which restic uses by default) was never affected, and even if you used
+`max` compression, chances of being affected by this issue were small.
+
+To check a repository for any corruption, run `restic check --read-data`. This
+will download and verify the whole repository and can be used at any time to
+completely verify the integrity of a repository. If the `check` command detects
+anomalies, follow the suggested steps.
+
+https://github.com/restic/restic/issues/4677
+https://github.com/restic/restic/pull/4679
```
Comment (rawtaz) [inline changelog/unreleased/issue-4677:19]: Bugfix: Downgrade zstd library to fix rare data corruption at max. compression

In restic 0.16.3, backups where the compression level was set to `max` (using
`--compression max`) could in rare and very specific circumstances result in
data corruption due to a bug in the library used for compressing data. Restic
0.16.1 and 0.16.2 were not affected.

Restic now uses the previous version of the library used to compress data, the
same version used by restic 0.16.2. Please note that the `auto` compression
level (which restic uses by default) was never affected, and even if you used
`max` compression, chances of being affected by this issue are small.

To check a repository for any corruption, run `restic check --read-data`. This
will download and verify the whole repository and can be used at any time to
completely verify the integrity of a repository. If the `check` command detects
anomalies, follow the suggested steps.

https://github.com/restic/restic/issues/4677
https://github.com/restic/restic/pull/4679

**Review by rawtaz — CHANGES_REQUESTED** [review]
Very minor change, otherwise good!

File: changelog/unreleased/issue-4677 (hunk 1 above)
Comment (MichaelEischer) [reply changelog/unreleased/issue-4677]: Thanks!

**Review by rawtaz — APPROVED** [review]
LGTM

**Pull request merged**